core/src/main/java/io/kestra/core/schedulers/AbstractScheduler.java

```diff
@@ -122,27 +117,60 @@ public void run() {
         );
         thread.start();
 
-        // remove trigger on flow update
+        // remove trigger on flow update & update local triggers store
         this.flowListeners.listen((flow, previous) -> {
-            synchronized (triggerStateSavedLock) {
-                if (flow.isDeleted()) {
-                    ListUtils.emptyOnNull(flow.getTriggers())
-                        .forEach(abstractTrigger -> {
-                            Trigger trigger = Trigger.of(flow, abstractTrigger);
-
-                            triggerStateSaved.remove(trigger.uid());
-                            triggerQueue.delete(trigger);
-                        });
-                } else if (previous != null) {
-                    FlowService
-                        .findRemovedTrigger(flow, previous)
-                        .forEach(abstractTrigger -> {
-                            Trigger trigger = Trigger.of(flow, abstractTrigger);
-
-                            triggerStateSaved.remove(trigger.uid());
-                            triggerQueue.delete(trigger);
-                        });
-                }
+            if (flow.isDeleted() || previous != null) {
+                List<AbstractTrigger> triggersDeleted = flow.isDeleted() ? ListUtils.emptyOnNull(flow.getTriggers()) : FlowService
+                    .findRemovedTrigger(flow, previous);
+                triggersDeleted.forEach(abstractTrigger -> {
+                    Trigger trigger = Trigger.of(flow, abstractTrigger);
+                    this.triggerQueue.delete(trigger);
+                });
+            }
+            if (previous != null) {
+                FlowService.findUpdatedTrigger(flow, previous)
+                    .forEach(abstractTrigger -> {
+                        if (abstractTrigger instanceof PollingTriggerInterface) {
+                            RunContext runContext = runContextFactory.of(flow, abstractTrigger);
+                            ConditionContext conditionContext = conditionService.conditionContext(runContext, flow, null);
+                            Optional<Trigger> lastTrigger = this.triggerState.findLast(Trigger.of(flow, abstractTrigger));
+                            try {
+                                Trigger updatedTrigger = Trigger.builder()
+                                    .tenantId(flow.getTenantId())
+                                    .namespace(flow.getNamespace())
+                                    .flowId(flow.getId())
+                                    .flowRevision(flow.getRevision())
+                                    .triggerId(abstractTrigger.getId())
+                                    .date(now())
+                                    .nextExecutionDate(((PollingTriggerInterface) abstractTrigger).nextEvaluationDate(conditionContext, lastTrigger))
+                                    .build();
+                                this.triggerState.update(updatedTrigger);
```

Comment:
I have a concern here, I think there is a race.
It's possible that you find a trigger via `findLast` which is updated by the main scheduler loop before you save it.